checkstyle-tester/README.md

```diff
@@ -81,25 +81,59 @@ When the script finishes its work the following directory structure will be crea
 
 You will find *index.html* file in /reports/diff directory. The file represents summary diff report.
 
-To generate a diff report before and after your changes manually please use the tool
-[https://github.com/attatrol/ahsm](https://github.com/attatrol/ahsm)
-
 ATTENTION: 
 
+Administrators recommend you modify `projects-to-test-on.properties` and test as many projects as possible. Each project has its own unique style and it is common to find regression in 1 and not the others.
+
 you can specify at projects-to-test-on.properties path to local file system if you have some 
 repositories on your local, example: "checkstyle=/home/username/java/git-repos/checkstyle/checkstyle".
```

Comment:
I don't think this information is still correct.
At least the format is not correct.
Despite the file still being named `....properties`, it is not a properties file (anymore I guess).
Also, I was not able to get this working somehow and the code does not look like it would be a supported case.
Not even `github` as type that is mentioned in the `properties` file header seems to be supported, only `git` and `hg` currently.
If the local code actually is a Git repository, one can use a local path or a file URL with the Git type to make it work the Git way.
But besides that the example is wrong, it would be much nicer if locally available code is just supported, this way it would work with any locally available code from any VCS including none and it would also save space and time as you don't have to re-clone the repository but can just copy the to-be-tested files from the locally available code.
I'll open another ticket to properly support local directories, but at least this part of the docs is not correct. The `on your local` is missing an object and the example is not working.

Reply:
> it is not a properties file 

True, but it may be too late to change this. It also won't be done in this PR which is just updating and expanding documentation, not changing logic.

> Also, I was not able to get this working somehow and the code does not look like it would be a supported case.
> The on your local is missing an object and the example is not working.

Please create an issue. Once it is working, we will create a valid example from that. We can't do it before, because we might have to add new types to support it.

> you don't have to re-clone the repository but can just copy the to-be-tested files from the locally available code.

Once the repo is cloned, it shouldn't be deleting the repo. It can continuously be re-used over and over for each new regression. Thats why we have a existence check before cloning.
https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/launch.groovy#L119

Reply:
> True, but it may be too late to change this. It also won't be done in this PR which is just updating and expanding documentation, not changing logic.

I didn't suggest this, I just mentioned it as indication that the syntax of the example still assumes it is a proper properties file which it is not anymore.

> Please create an issue.

As I said, I will. :-)

> Once it is working, we will create a valid example from that. We can't do it before, because we might have to add new types to support it.

As I said, if it actually is a local Git repository you can make it work with path or file-url. And you could add an example for this way and if it is possible with HG then an example for that too, or you could remove the example and the sentence, as just specifying a local path currently does not work.

> > you don't have to re-clone the repository but can just copy the to-be-tested files from the locally available code.
> 
> Once the repo is cloned, it shouldn't be deleting the repo. It can continuously be re-used over and over for each new regression. Thats why we have a existence check before cloning.
> https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/launch.groovy#L119

I didn't mean re-cloned when it is already in `repositories`.
But if you currently have a local Git repo and use a local path or file-URL, it will be cloned into `repositories` while everything would have already been available on the local disk.
This is what I called re-cloned in this case and that could be saved when a local path is simply supported without VCS-dependency. But that is for the other issue to come actually, not for this one.

Reply:
Sorry if missed some things on what you were saying.

> if it actually is a local Git repository you can make it work with path or file-url

I added this example. So it is non-git/hg projects you are looking to add?

Reply:
Better, thanks, just one nit if you allow: On your local what?
Yes, basically any project that lies on the local disk could be supported easily.
And even projects that are local git or hg projects could just be used as is.
This would allow to use the project even with local modifications, it allows to use the state that you have currently on your disk and it saves the space of duplicating the project into `repositories` folder.
I'm playing around with it currently and will then open it as issue.

Reply:
Fixed the wording.
`You can specify at projects-to-test-on.properties, a path to a local file system's git or hg repository to run regressions on any projects you have personally saved.`.